File: Go/Range_Addition.go
func getModifiedArray(length int, updates [][]int) []int {
    incs := make([]int, length)
    start,end,val := 0, 0, 0
    for i := range updates {
        start = updates[i][0]
        end = updates[i][1]
        val = updates[i][2]
        incs[start] += val
        if end < length-1 {
            incs[end+1] -= val
        }
    }
    for i := 1; i < length; i++ {
        incs[i] += incs[i-1]
    }
    return incs
}

File: Python/Range_Addition.py
class Solution(object):
    def getModifiedArray(self, length, updates):
        """
        :type length: int
        :type updates: List[List[int]]
        :rtype: List[int]
        """
        # store the sums to be used by its self and following elements
        incs = [0]*length
        for update in updates:
            x,y,val = update[0],update[1],update[2]
            incs[x] += val
            if y < length - 1:
                incs[y+1] -= val
        for i in range(1, length):
            incs[i] += incs[i-1]

        return incs
